src/cargo-fmt/main.rs

```diff
@@ -748,4 +861,197 @@ mod cargo_fmt_tests {
             );
         }
     }
+
+    mod get_nested_integration_test_files_tests {
```

Comment:
As this new behavior is on the `cargo fmt` side determining the correct set of files to pass to `rustfmt`, I wanted to ensure we had some automated tests. I decided to create some mini-sample projects under the `tests` directory and then adding some tests here against the functions that are determining the targets/files.